### PR TITLE
imposm3 0.11.1 (new formula)

### PR DESCRIPTION
**Pull request by trytuna**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by trytuna** [comment]
Can someone help me to fix this issue? It ran fine locally with homebrew.

However I ran into this [this](https://github.com/omniscale/imposm3/issues/132#issuecomment-1086929296) issue while compiling imposm3 from source without homebrew.

**Comment by SMillerDev** [comment]
It seems that the things you declared a build dependency are actually runtime dependencies. Try removing them and running `brew test imposm3` to check

**Comment by SMillerDev** [comment]
I'm not arguing you need it to build. But the current syntax you're using `depends_on "stuff" => :build` means it's _only_ needed to build, which the errors seemed to indicate was not the case.

Remove the `=> :build` to make a it a runtime and test dependency too

**Comment by trytuna** [comment]
Got you. Thanks for clarifying that. I've updated it accordingly

File: Formula/imposm3.rb
Comment (SMillerDev) [inline Formula/imposm3.rb:21]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: Formula/imposm3.rb
Comment (trytuna) [reply Formula/imposm3.rb:13]: Would nice very clean if this works. Unfortunately `ld` doesn't find leveldb:

```
==> go build -trimpath -o=/opt/homebrew/Cellar/imposm3/0.11.1/bin/imposm -ldflags=-X github.com/omniscale/imposm3.Version=0.11.1 cmd/imposm/main.go
# command-line-arguments
/opt/homebrew/Cellar/go/1.18/libexec/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
ld: library not found for -lleveldb
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

File: Formula/imposm3.rb
Comment (trytuna) [reply Formula/imposm3.rb]: I've done that. Makes it much simpler 👌

File: Formula/imposm3.rb
Comment (trytuna) [reply Formula/imposm3.rb:21]: Testing the functionality of this tool requires a protobuf file (which is binary). I could download it from _any_ source while testing but this add a crucial dependency to this package. What would be the preferred way to solve this on your side?

File: Formula/imposm3.rb
Comment (trytuna) [reply Formula/imposm3.rb]: Works like a charm 😇

File: Formula/imposm3.rb
Comment (SMillerDev) [reply Formula/imposm3.rb:21]: See the `resource` blocks in other formula for a way to downloade extra files in a formula

Comment (trytuna) [reply Formula/imposm3.rb:51]: I think I came up with an even better solution. See last commit

**Review by SMillerDev — DISMISSED** [review]

**Comment by SMillerDev** [comment]
Thanks @trytuna  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by SMillerDev** [comment]
@trytuna can you rebase this and squash all your commits into one? The CI is having trouble merging the current commits

**Comment by trytuna** [comment]
> @trytuna can you rebase this and squash all your commits into one? The CI is having trouble merging the current commits

It should be all set now - sorry for the inconvenience

**Review by SMillerDev — APPROVED** [review]

**Comment by trytuna** [comment]
Do I have to do something to get it merged or do we wait for another review?

**Pull request closed without merging**
